src/BDInfo/TSPlaylistFile.cs

```diff
@@ -17,1346 +17,1150 @@
+            {
+                Chapters.Add(newClip.RelativeTimeIn);
             }
-            LoadStreamClips();
-            IsInitialized = true;
         }
+        LoadStreamClips();
+        IsInitialized = true;
+    }
 
-        public override string ToString()
-        {
-            return Name;
-        }
+    public override string ToString()
+    {
+        return Name;
+    }
 
-        public ulong InterleavedFileSize
+    public ulong InterleavedFileSize
+    {
+        get
         {
-            get
-            {
-                ulong size = 0;
-                foreach (TSStreamClip clip in StreamClips)
-                {
-                    size += clip.InterleavedFileSize;
-                }
-                return size;
-            }
+            return StreamClips.Aggregate<TSStreamClip, ulong>(0, (current, clip) => current + clip.InterleavedFileSize);
         }
-        public ulong FileSize
+    }
+    public ulong FileSize
+    {
+        get
         {
-            get
-            {
-                ulong size = 0;
-                foreach (TSStreamClip clip in StreamClips)
-                {
-                    size += clip.FileSize;
-                }
-                return size;
-            }
+            return StreamClips.Aggregate<TSStreamClip, ulong>(0, (current, clip) => current + clip.FileSize);
         }
-        public double TotalLength
+    }
+    public double TotalLength
+    {
+        get
         {
-            get
-            {
-                double length = 0;
-                foreach (TSStreamClip clip in StreamClips)
-                {
-                    if (clip.AngleIndex == 0)
-                    {
-                        length += clip.Length;
-                    }
-                }
-                return length;
-            }
+            return StreamClips.Where(clip => clip.AngleIndex == 0).Sum(clip => clip.Length);
         }
+    }
 
-        public double TotalAngleLength
+    public double TotalAngleLength
+    {
+        get
         {
-            get
-            {
-                double length = 0;
-                foreach (TSStreamClip clip in StreamClips)
-                {
-                    length += clip.Length;
-                }
-                return length;
-            }
+            return StreamClips.Sum(clip => clip.Length);
         }
+    }
 
-        public ulong TotalSize
+    public ulong TotalSize
+    {
+        get
         {
-            get
-            {
-                ulong size = 0;
-                foreach (TSStreamClip clip in StreamClips)
-                {
-                    if (clip.AngleIndex == 0)
-                    {
-                        size += clip.PacketSize;
-                    }
-                }
-                return size;
-            }
+            return StreamClips.Where(clip => clip.AngleIndex == 0).Aggregate<TSStreamClip, ulong>(0, (current, clip) => current + clip.PacketSize);
```

Comment:
Why not use Sum here?

Reply:
Because of ambiguous overloads and `ulong`